matrix-sdk/src/main/java/org/matrix/androidsdk/util/ResourceUtils.java

```diff
@@ -150,7 +150,7 @@ public static Bitmap createThumbnailBitmap(Context context, Uri mediaUri, int ma
 
                 // the bitmap is smaller that max sizes
                 if ((fullSizeBitmap.getHeight() < maxThumbHeight) && (fullSizeBitmap.getWidth() < maxThumbWidth)) {
-                    thumbnailBitmap = fullSizeBitmap;
+                    thumbnailBitmap = fullSizeBitmap.copy(Bitmap.Config.ARGB_8888, true);
```

Comment:
So right here I think the bitmap was getting passed by reference. When fullSizeBitmap is recycled below thumbnailBitmap also breaks, which causes the error you fixed in the MediaCache. This line fixes the root cause, but since the system runs its own gc we might as well throw away the manual recycle below as well as the other line that copies the bitmap in MediaCache.